python-avd/pyavd/_eos_designs/structured_config/overlay/router_adaptive_virtual_topology.py

```diff
@@ -41,7 +43,10 @@ def router_adaptive_virtual_topology(self: AvdStructuredConfigOverlayProtocol) -
             raise AristaAvdInvalidInputsError(msg)
 
         # Edge or Transit
-        self.structured_config.router_adaptive_virtual_topology._update(topology_role=self.shared_utils.cv_pathfinder_role, zone=self.shared_utils.wan_zone)
+        self.structured_config.router_adaptive_virtual_topology._update(
+            topology_role=cast("Literal['edge', 'pathfinder', 'transit region', 'transit zone']", self.shared_utils.cv_pathfinder_role),
```

Comment:
is this the right thing to do? We could rather change the return type of self.shared_utils.cv_pathfinder_role to be `Literal['edge', 'pathfinder', 'transit region', 'transit zone'] | None` and replace line 27 above by

```
if not self.shared_utils.is_cv_pathfinder_role:`
```

Reply:
Good point. Updated the shared_util. No need to check for None since None is also a valid value for this field in the output (will never happen but type checker is happy)